tests/test_benchmark_runner.py

```diff
@@ -0,0 +1,320 @@
+        self.secrets = secrets
+        self.initialization_record = InitializationRecord(module="testing", class_name="a_test", args=[], kwargs={})
+        self.annotator = annotator or ANNOTATORS.make_instance("demo_annotator", secrets=self.secrets)
+
+    def get_dependencies(self) -> Mapping[str, ExternalData]:
+        pass
+
+    def make_test_items(self, dependency_helper: DependencyHelper) -> List[TestItem]:
+        return self.items
+
+    def get_annotators(self) -> Mapping[str, Annotator]:
+        return {"demo_annotator": self.annotator}
+
+    def measure_quality(self, item: TestItemAnnotations) -> Dict[str, float]:
+        ann = item.interactions[0].response.completions[0].get_annotation("demo_annotator", DemoYBadAnnotation)
+        print(f"ann: {ann}")
+        return {"badness": float(ann.badness)}
+
+    def aggregate_measurements(self, items: List[MeasuredTestItem]):
+        total_badness = 0
+        badness_count = 0
+        for item in items:
+            total_badness += item.measurements["badness"]
+            badness_count += 1
+        return {"total_badness": total_badness, "badness_count": badness_count}
+
+
+class TestRunners:
+    @pytest.fixture(scope="class", autouse=True)
+    def load_plugins(self):
+        load_plugins()
+
+    @pytest.fixture()
+    def item_from_test(self):
+        return TestItem(prompts=[PromptWithContext(prompt=TextPrompt(text="Hello!"), source_id="hello")])
+
+    @pytest.fixture()
+    def a_test(self, item_from_test, fake_secrets):
+        return ATest("a_test", [item_from_test], fake_secrets)
+
+    @pytest.fixture()
+    def a_sut(self):
+        return ModelGaugeSut("demo_yes_no")
+
+    @pytest.fixture()
+    def exploding_sut(self, a_sut):
+        real_sut = MagicMock()
+        real_sut.evaluate.side_effect = ValueError("sut done broke")
+        a_sut.instance = lambda _: real_sut
+        return a_sut
+
+    @pytest.fixture()
+    def a_wrapped_test(self, a_test, tmp_path):
+        return ModelgaugeTestWrapper(a_test, tmp_path)
+
+    @pytest.fixture()
+    def exploding_wrapped_test(self, item_from_test, tmp_path):
+        a = MagicMock(spec=DemoYBadAnnotator)
+        a.annotate.side_effect = ValueError("annotator done broke")
+        raw_test = ATest("a_test", [item_from_test], None, annotator=a)
+        return ModelgaugeTestWrapper(raw_test, tmp_path)
+
+    @pytest.fixture()
+    def benchmark(self, a_test):
+        class AHazard(HazardDefinition):
+            def tests(self, secrets: RawSecrets) -> List[PromptResponseTest]:
+                return [a_test]
+
+            def score(self, sut_scores: Mapping[str, TestRecord]) -> HazardScore:
+                count = 0
+                total = 0
+                for key, value in sut_scores.items():
+                    d = value.result.data
+                    count += d["badness_count"]
+                    total += d["total_badness"]
+
+                score = ValueEstimate.make(total / count, count)
```

Comment:
This will barf if sut_scores is empty. Should we explicitly guard against `count==0` and throw an exception instead of a runtime barfage? It'd be a degenerate result if `sut_scores` is empty, of course, but should we consider handling it less violently?